### PR TITLE
Chore: Fail-fast on callbacks

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

The idea of this PR is to skip some omnichannel-only callbacks when the room/message being passed is not from Omnichannel. This will allow other messages to perform a little bit better.

### CODE REVIEW

File: apps/meteor/ee/app/livechat-enterprise/server/hooks/handleNextAgentPreferredEvents.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ const checkDefaultAgentOnNewRoom = (defaultAgent, defaultGuest) => {
 		return defaultAgent;
 	}
 
+	if (!lastChattedAgentPreferred) {
+		return defaultAgent;
+	}
+
```
Comment (murtaza98) [inline apps/meteor/ee/app/livechat-enterprise/server/hooks/handleNextAgentPreferredEvents.js:31]: This changes the current logic when "Contact manager preferred routing" is on but "last preferred agent routing" is off. Based on the current logic, it won't transfer the chat to the contactManager.

Maybe you can add a check for both settings here, like

```
if (!lastChattedAgentPreferred && !contactManagerPreferred) {
		return defaultAgent;
}
```

Comment (KevLehman) [reply apps/meteor/ee/app/livechat-enterprise/server/hooks/handleNextAgentPreferredEvents.js:31]: :thinking: I think I'll leave this one where it was before, since seems it affects some functionality

**Review by murtaza98 — APPROVED** [review]

**Comment by KevLehman** [comment]
@sampaiodiego , from our side, no changes would change "functionality", so we should be good skipping QA for this one after you review it :eyes:

File: apps/meteor/app/models/server/models/Subscriptions.js [hunk 2]
```diff
@@ -28,9 +28,9 @@ export class Subscriptions extends Base {
 		this.tryEnsureIndex({ desktopNotifications: 1 }, { sparse: 1 });
 		this.tryEnsureIndex({ mobilePushNotifications: 1 }, { sparse: 1 });
 		this.tryEnsureIndex({ emailNotifications: 1 }, { sparse: 1 });
+		this.tryEnsureIndex({ audioNotifications: 1 }, { sparse: 1 });
```
Comment (sampaiodiego) [inline apps/meteor/app/models/server/models/Subscriptions.js:31]: not sure where this came from 🤔

File: apps/meteor/app/models/server/models/Subscriptions.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ export class Subscriptions extends Base {
 	constructor(...args) {
 		super(...args);
 
-		this.tryEnsureIndex({ rid: 1 });
```
Comment (sampaiodiego) [inline apps/meteor/app/models/server/models/Subscriptions.js:15]: how are we going to remove this index later on? 👀

File: apps/meteor/app/lib/server/lib/sendNotificationsOnMessage.js [hunk 4]
```diff
@@ -372,8 +372,11 @@ export async function sendAllNotifications(message, room) {
 			}),
 		)
 			.then((users) => {
+				const subscriptions = Subscriptions.findByRoomIdAndUserIds(room._id, users, {
+					fields: { 'u._id': 1 },
+				});
```
Comment (sampaiodiego) [inline apps/meteor/app/lib/server/lib/sendNotificationsOnMessage.js:377]: you need all fields here
```suggestion
				const subscriptions = Subscriptions.findByRoomIdAndUserIds(room._id, users);
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/app/models/server/models/Subscriptions.js (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/app/models/server/models/Subscriptions.js:31]: Hmmm, copied this from the old PR. I kinda remember this prop was used on "notifications", but cannot find any ref to it anymore

File: apps/meteor/app/models/server/models/Subscriptions.js (hunk 3 above)
Comment (KevLehman) [reply apps/meteor/app/models/server/models/Subscriptions.js:15]: I have the migration, but migrations are a sin now, so not sure if I should sin :trollface:

Comment (sampaiodiego) [reply apps/meteor/app/models/server/models/Subscriptions.js:15]: we got rid of this already 😬

File: apps/meteor/app/models/server/models/Subscriptions.js [hunk 5]
```diff
@@ -12,8 +12,8 @@ export class Subscriptions extends Base {
...
 		this.tryEnsureIndex({ rid: 1, ls: 1 });
+		this.tryEnsureIndex({ 'userHighlights.0': 1, 'rid': 1 });
```
Comment (sampaiodiego) [inline apps/meteor/app/models/server/models/Subscriptions.js:16]: what if you revert the changes to indexes and we do them in another PR?

Comment (KevLehman) [reply apps/meteor/app/models/server/models/Subscriptions.js:16]: Fair :eyes:

File: apps/meteor/app/models/server/models/Subscriptions.js [hunk 6]
```diff
@@ -30,7 +30,6 @@ export class Subscriptions extends Base {
 		this.tryEnsureIndex({ emailNotifications: 1 }, { sparse: 1 });
 		this.tryEnsureIndex({ autoTranslate: 1 }, { sparse: 1 });
 		this.tryEnsureIndex({ autoTranslateLanguage: 1 }, { sparse: 1 });
-		this.tryEnsureIndex({ 'userHighlights.0': 1 }, { sparse: 1 });
```
Comment (sampaiodiego) [inline apps/meteor/app/models/server/models/Subscriptions.js:33]: still removing an index 🙈

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

**Review by sampaiodiego — APPROVED** [review]

File: apps/meteor/app/models/server/models/Subscriptions.js (hunk 6 above)
Comment (KevLehman) [reply apps/meteor/app/models/server/models/Subscriptions.js]: Wait, i removed that one lol. Thanks for fixing it.
